daemon/buildinfo.c

```diff
@@ -6,7 +6,11 @@
 // Optional features
 
 #ifdef ENABLE_ACLK
-#define FEAT_CLOUD "YES"
+#ifdef ACLK_NG
+#define FEAT_CLOUD "YES (Next Generation)"
```

Comment:
We should probably come up with a name other than ‘next generation’ for this. Perhaps we should just start versioning the low-level transport implementation?

Reply:
We could change that in the next PRs I guess. My plan was after a time Legacy will be removed completely and this just becomes "ACLK" without additional designations

Reply:
Yeah, that does make sense. I just want to avoid having generic naming bite us in the arse down the road.

On a side note, the version that completely removes the old transport layer should probably be a major version bump. Regardless of whether we have an on-premises Cloud offering at that point or not, it’s a breaking change and should be handled as such.

Reply:
I generally agree with you but would think about this as separate PR + we need to keep a way, not to only version ACLK but possibly have alternative implementation ACLK Legacy/NG Timo/NG Paho.... so it is not always a newer version only currently Legacy and NG are on-par feature-wise.

Reply:
In that case, perhaps ‘ACLK (simplified)’ for this one? AFAICT this new implementation is significantly simpler than the legacy implementation that needs mosquitto and LWS.

Reply:
Well, I do not object to renaming if consensus is found but would not like to mess up this PR as it is huge already. But open to doing that in the future.